src/client/subject.ts

```diff
@@ -0,0 +1,310 @@
+import EppoClient, { IAssignmentDetails, IContainerExperiment } from './eppo-client';
+import { Attributes, BanditActions, ContextAttributes, FlagKey } from '../types';
+import { ensureNonContextualSubjectAttributes } from '../attributes';
+import { Configuration } from '../configuration';
+
+/**
+ * A wrapper around EppoClient that automatically supplies subject key, attributes, and bandit
+ * actions for all assignment and bandit methods.
+ *
+ * This is useful when you always want to use the same subject and attributes for all flag
+ * evaluations.
+ */
+export class Subject {
```

Comment:
This is subject-aware client

Initially named it subject-bound client but then was debating with myself whether I should duplicate *ALL* public API from client or just evaluation methods. We can always expose more methods later, so decided to only implement evaluation ones. Then realized that we can name it "Subject" and have `getPrecomputedSubject()` on EppoClient, so that seemed like a neat idea